edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/WorkerTests.cs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.MetricsCollector.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Moq;
+    using Xunit;
+
+    public class WorkerTests : TempDirectory
```

Comment:
This is a bit odd.. create an instance in this object instead

Reply:
I did it like this to take advantage of the automatic IDisposable. I can use composition instead